lib/cusparse/preconditioners.jl

```diff
@@ -280,3 +294,49 @@ for elty in (:Float32, :Float64, :ComplexF32, :ComplexF64)
         end
     end
 end
+
+# gtsv2
+for (bname_pivot,fname_pivot,bname_nopivot,fname_nopivot,elty) in ((:cusparseSgtsv2_bufferSizeExt, :cusparseSgtsv2, :cusparseSgtsv2_nopivot_bufferSizeExt, :cusparseSgtsv2_nopivot, :Float32),
+                                                                   (:cusparseDgtsv2_bufferSizeExt, :cusparseDgtsv2, :cusparseDgtsv2_nopivot_bufferSizeExt, :cusparseDgtsv2_nopivot, :Float64),
+                                                                   (:cusparseCgtsv2_bufferSizeExt, :cusparseCgtsv2, :cusparseCgtsv2_nopivot_bufferSizeExt, :cusparseCgtsv2_nopivot, :ComplexF32),
+                                                                   (:cusparseZgtsv2_bufferSizeExt, :cusparseZgtsv2, :cusparseZgtsv2_nopivot_bufferSizeExt, :cusparseZgtsv2_nopivot, :ComplexF64))
+    @eval begin
+        function gtsv2!(dl::CuVector{$elty}, d::CuVector{$elty}, du::CuVector{$elty}, B::CuVecOrMat{$elty}, index::SparseChar='O'; pivoting::Bool=true)
+            ml = length(dl)
+            m = length(d)
+            mu = length(du)
+            mB = size(B,1)
+            (m ≤ 2) && throw(DimensionMismatch("The size of the linear system must be at least 3."))
+            !(ml == m == mu) && throw(DimensionMismatch("(dl, d, du) must have the same length, the size of the vectors is ($ml,$m,$mu)!"))
+            (m != mB) && throw(DimensionMismatch("The tridiagonal matrix and the right-hand side B have inconsistent dimensions ($m != $mB)!"))
+            n = size(B,2)
+            ldb = max(1,stride(B,2))
+
+            function bufferSize()
+                out = Ref{Csize_t}(1)
+                if pivoting
+                    $bname_pivot(handle(), m, n, dl, d, du, B, ldb, out)
+                else
+                    $bname_nopivot(handle(), m, n, dl, d, du, B, ldb, out)
+                end
+                return out[]
+            end
+            with_workspace(bufferSize) do buffer
+                if pivoting
+                    $fname_pivot(handle(), m, n, dl, d, du, B, ldb, buffer)
+                else
+                    $fname_nopivot(handle(), m, n, dl, d, du, B, ldb, buffer)
+                end
+            end
+            B
+        end
+    end
+end
+
+for elty in (:Float32, :Float64, :ComplexF32, :ComplexF64)
+    @eval begin
+        function gtsv2(dl::CuVector{$elty}, d::CuVector{$elty}, du::CuVector{$elty}, B::CuVecOrMat{$elty}, index::SparseChar='O'; pivoting::Bool=true)
+            gtsv2!(dl, d, du, copy(B), index, pivoting=pivoting)
```

Comment:
Here, and in other places:

```suggestion
            gtsv2!(dl, d, du, copy(B), index; pivoting)
```